catalyst_voices/packages/libs/catalyst_cose/lib/src/types/cose_headers.dart

```diff
@@ -161,6 +169,28 @@ final class CoseHeaders extends Equatable {
     }
   }
 
+  /// Returns a copy of the [CoseHeaders] with given [alg] and [kid].
+  CoseHeaders copyWith({
+    required StringOrInt? alg,
+    required Uint8List? kid,
+  }) {
+    return CoseHeaders(
+      alg: alg,
+      kid: kid,
```

Comment:
don't you want current value as fallback?

```dart
      alg: alg ?? this.alg,
      kid: kid ?? this.kid,
```

Reply:
I want to override both without a fallback. Both fields are nullable so I would need something like `Optional`. Though `catalyst_cose` is a published package so it cannot depend on non-published (internal) package. Do you think I should duplicate the `Optional` class for this package or leave it as it is?